vm_insnhelper.c

```diff
@@ -4675,15 +4668,7 @@ vm_opt_newarray_max(rb_execution_context_t *ec, rb_num_t num, const VALUE *ptr)
 	}
     }
     else {
-	VALUE ary = rb_ary_new4(num, ptr);
-        const rb_callable_method_entry_t *me =
-            rb_callable_method_entry_with_refinements(rb_cArray, idMax, NULL);
-        if (me) {
-            return rb_vm_call0(ec, ary, idMax, 0, NULL, me, RB_NO_KEYWORDS);
-        }
-        else {
-            return rb_funcall(ary, idMax, 0);
-        }
+        return rb_vm_call_with_refinements(ec, rb_ary_new4(num, ptr), idMax, 0, NULL, RB_NO_KEYWORDS);
```

Comment:
I thought the same thing too, and found an `ASSUME` would make the generated code better.

```suggestion
	VALUE ary = rb_ary_new4(num, ptr);
	ASSUME(! SPECIAL_CONST_P(ary));
        return rb_vm_call_with_refinements(ec, ary, idMax, 0, NULL, RB_NO_KEYWORDS);
```

Same for `vm_opt_newarray_min`.

Reply:
Hmmm, what I tried was the `static inline` in vm_insnhelper.c.
May not be effective to `MJIT_FUNC_EXPORTED` function.

Reply:
OK.  I'm going to merge this as it is.  If you think `static inline` or `ASSUME` would be helpful, please add them.  Thank you for your review.